### PR TITLE
cache get_all_xmlns_app_id_pairs_submitted_to_in_domain

### PR DESCRIPTION
**Pull request by snopoke**
What do you think of this?

### CODE REVIEW

**Review by jmtroth0 — APPROVED** [review]
Seems reasonable to me. Is there any way to find some data that could back that up?

I also don't have an instinct for ES yet: Is this a particularly long running query?

File: corehq/ex-submodules/couchforms/analytics.py [hunk 1]
```diff
@@ -77,7 +77,16 @@ def app_has_been_submitted_to_in_last_30_days(domain, app_id):
...
+@quickcache(['domain'], memoize_timeout=0, timeout=12 * 3600)
 def get_all_xmlns_app_id_pairs_submitted_to_in_domain(domain):
+    """This is used to get (XMLNS, app_id) from submitted forms. The results
+    get combined with data from all current app versions which means
+    that this is only used to get (XMLNS, app_id) combos from forms submitted
+    in the past which no longer have a corresponding form in the app (e.g. form deleted)
+
+    Given that we can cache this for a long period of time under the assumption that
+    a user isn't going to submit a form and then delete it from their app immediately.
+    """
```
Comment (dannyroberts) [inline corehq/ex-submodules/couchforms/analytics.py]: Haha, I have to say... _someone_ is going to do this and then be really confused. What if deleting a form / module / app (/ build?) invalidated this cache (for the whole domain)? I _think_ it's simple enough to implement (though cache invalidation always starts out _seeming_ simple...), but I'll just leave it up to you to decide.

**Comment by snopoke** [comment]
@jmtroth0 do you mean data about users deleting a form? I don't think we have that.

@dannyroberts fair point. I'll take a look at that.

File: corehq/ex-submodules/couchforms/analytics.py (hunk 1 above)
Comment (snopoke) [reply corehq/ex-submodules/couchforms/analytics.py]: 1333aee

**Review by dannyroberts — APPROVED** [review]
Looking sharp!

**Pull request merged**